torch/_inductor/ir.py

```diff
@@ -2029,6 +2030,181 @@ def realize(self):
+
+    # TODO mlazos: Ensure each Bufferlist has uniform dtype and device
+    def get_device(self, ind=0):
+        return self.data.layouts[ind].device
+
+    # TODO mlazos: Ensure each Bufferlist has uniform dtype and device
+    def get_dtype(self, ind=0):
+        return getattr(self.data.layouts[ind], "dtype", None)
+
+    def get_size(self, ind):
+        return list(self.data.layouts[ind].size)
+
+    def get_stride(self, ind):
+        return list(self.data.layouts[ind].stride)
+
+    def get_layout(self, ind):
+        return self.data.layouts[ind]
+
+    def get_layouts(self):
+        return self.data.layouts
+
+    def get_storage_numel(self):
+        return self.get_numel()
+
+    def is_extern(self):
+        return False
+
+    def freeze_layout(self):
+        if not isinstance(self.layout, (MultiOutputLayout, AliasedLayout)):
+            self.layout = self.layout.as_fixed()
+
+    def freeze_layout_with_stride_order(self, order):
+        assert isinstance(self.layout, FlexibleLayout)
+        self.layout = self.layout.as_stride_order(order)
+
+    def freeze_layout_with_fill_order(self, order):
+        assert isinstance(self.layout, FlexibleLayout)
+        self.layout = self.layout.as_fill_order(order)
+
+    def freeze_layout_with_same_order(self, stride):
+        assert isinstance(self.layout, FlexibleLayout)
+        self.layout = self.layout.as_same_order(stride)
+
+    def make_loader(self):
+        def fn():
+            load = self.data.make_loader()
+            return load()
+
+        return fn
+
+    def get_store_function(self):
+        def fn(result):
+            ops.store(self.list_name, None, result)
+
+        return fn
+
+    def is_no_op(self):
+        return False
+
+    def codegen_reference(self):
+        return self.get_name()
+
+    def decide_layout(self):
+        pass
+
+    def get_alias_names(self):
+        return ()
+
+    def get_mutation_names(self):
+        return ()
+
+    def list_arg_count(self):
+        _, (list_reads, list_writes) = self.extract_read_writes()
+        return len(list_reads) + len(list_writes)
+
+    @cache_on_self
+    def normalized_read_writes(self):
+        return self.extract_read_writes()[0]
+
+    @cache_on_self
+    def extract_read_writes(self):
+        list_load_extractor = dependencies.RecordListLoadStores()
+
+        with V.set_ops_handler(list_load_extractor):
+            self.get_store_function()(self.make_loader()())
+
+        return (
+            dependencies.ReadWrites(
+                list_load_extractor._reads, list_load_extractor._writes, set()
+            ),
+            (list_load_extractor._list_reads, list_load_extractor._list_writes),
+        )
+
+    def get_read_writes(self):
+        return self.normalized_read_writes()
+
+    def realize(self):
+        pass
+
+    def _index(self, ind):
+        return self.data._index(ind)
+
+    def body(self, indices):
+        self.data.body(indices)
+
+
+class ListElemBuffer(Buffer):
```

Comment:
Try using a ComputedBuffer here isntead

Reply:
There was more complexity here since we didn't have a loops to pass as data - it felt weird to hack around it when it didn't really add anything, since I was still going to override the same methods that were already implemented and not take advantage of the parent class.